src/pickup.cpp

```diff
@@ -187,6 +187,10 @@ bool pick_one_up( const tripoint &pickup_target, item &newit, vehicle *veh,
             return false;
         }
     }
+    if( newit.has_flag( "MARKER" ) ) {
```

Comment:
instead of checking for the `MARKER` flag everywhere, how about just checking if the item type is `partial_construction`.  You could even make an `item::is_marker()` function.